livre/core/pipeline/Executable.h

```diff
@@ -0,0 +1,128 @@
+/* Copyright (c) 2011-2016, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _Executable_h_
+#define _Executable_h_
+
+#include <livre/core/types.h>
+#include <livre/core/pipeline/Future.h>
+
+namespace livre
+{
+
+/**
+ * The Executable class is constructed using ExecutableImpl
+ * classes. It wraps the methods for execution of ExecutableImpl.
+ * a ExecutableImpl. It provides extra pre/post
+ * conditions as futures to retrieve the current situation.
+ *
+ * According to the preconditions or postconditions,
+ * executors can decide on scheduling algorithms.
+ */
+class Executable
```

Comment:
I don't like very much what you've done here. You've encapsulated a shared pointer around a class that forwards the methods to the contained pointer. From a conceptual point of view, there's no difference between this class and ExecutableImplPtr. If this the way you've removed the ExecutablePtr objects from the the interface, I think it needs to be thought again, because you haven't really removed the pointer, just put some make up to make it look different (including all the dangers it implies, for example the day after you could wake up with a pointer by your side despite you thought it was a plain by value object)